src/object/object_primitive.c

```diff
@@ -16374,3 +16381,225 @@ mr_get_compression_length (const char *string, int charlen)
 
   return length;
 }
+
+
+/*
+ * mr_get_size_and_write_string_to_buffer ()
+ *	  			  : Writes a VARCHAR or VARNCHAR to buffer, without needing the buffer initialized.
+ *				    
+ * buf(out)			  : Buffer to be written to.
+ * val_p(in)			  : Memory area to be written to.
+ * value(in)			  : DB_VALUE to be written.
+ * val_size(out)		  : Disk size of the DB_VALUE.
+ * align(in)			  : 
+ *
+ *  Note:
+ *	We use this to avoid double compression when it is required to have the size of the DB_VALUE, previous
+ *	to the write of the DB_VALUE in the buffer.
+ */
+
+int
+mr_get_size_and_write_string_to_buffer (OR_BUF * buf, char *val_p, DB_VALUE * value, int *val_size, int align)
```

Comment:
@lucianlazar2211 

I'd like to propose to rename some functions because `mr_` functions are member functions of `PR_TYPE`.
`pr_` would be better. 

Please rename functions:
`mr_get_compressed_data_from_buffer` as `pr_get_compressed_data_from_buffer`
`mr_get_compression_length` as `pr_get_compression_length`
`mr_get_size_and_write_string_to_buffer` as `pr_get_size_and_write_string_to_buffer`
`mr_write_compressed_string_to_buffer` as `pr_write_compressed_string_to_buffer`